src/analysis/special_functions/log/deriv.lean

```diff
@@ -268,4 +268,57 @@ begin
       by simpa [pow_succ'] using mul_le_of_le_one_right (pow_nonneg (abs_nonneg x) i) (le_of_lt h) }
 end
 
+/-- Power series expansion of `log(1 + x) - log(1 - x)` for `|x| < 1`. -/
+lemma has_sum_log_sub_log_of_abs_lt_1 {x : ℝ} (h : |x| < 1) :
+  has_sum (λ k : ℕ, (2 : ℝ) * (1 / (2 * k + 1)) * x ^ (2 * k + 1)) (log (1 + x) - log(1 - x)) :=
+begin
+  have h₁ := has_sum_pow_div_log_of_abs_lt_1 h,
+  have h₂ := has_sum.mul_left (-1) (has_sum_pow_div_log_of_abs_lt_1 (eq.trans_lt (abs_neg x) h)),
+  rw [neg_one_mul, neg_neg, sub_neg_eq_add 1 x] at h₂,
+  let term := λ n : ℕ, (-1) * ((-x) ^ (n + 1) / ((n : ℝ) + 1)) + x ^ (n + 1) / (n + 1),
+  have h_term_eq_goal : term ∘ (*) 2 = λ k : ℕ, 2 * (1 / (2 * k + 1)) * x ^ (2 * k + 1),
+  { ext n,
+    dsimp [term],
+    rw [odd.neg_pow (⟨n, rfl⟩ : odd (2 * n + 1)) x],
+    push_cast,
+    ring_nf, },
+  rw [← h_term_eq_goal, (nat.mul_right_injective two_pos).has_sum_iff],
+  { exact h₂.add h₁, },
+  { intros m hm,
+    rw [range_two_mul, set.mem_set_of_eq] at hm,
+    refine (eq.refl ((-1) * ((-x) ^ (m + 1) / (↑m + 1)) + x ^ (m + 1) / (↑m + 1) = 0)).mpr _,
+    rw [even.neg_pow (nat.even_succ.mpr hm), nat.succ_eq_add_one, neg_one_mul, neg_add_self] },
+end
+
+/--
+A lemma expanding the log of a fraction. The goal is to use it in `power_series_log_succ_div`.
+-/
+lemma log_succ_div_eq_log_sub {a : ℝ} (h : 0 < a) :
+  log ((a + 1) / a) = log (1 + 1 / (2 * a + 1)) - log (1 - 1 / (2 * a + 1)) :=
+begin
+  have h₀ : (2 : ℝ) * a + 1 ≠ 0, by { linarith, },
+  have h₁ := h.ne',
+  rw ← log_div,
+  { apply congr_arg log,
+    field_simp,
+    linarith, },
+  { field_simp,
+    linarith } ,
+  { field_simp },
+end
+
+/-- Power series expansion of `log ((a + 1) / a)`. -/
+theorem power_series_log_succ_div {a : ℝ} (h : 0 < a) : has_sum (λ k : ℕ,
+  (2 : ℝ) * (1 / (2 * k + 1)) * (1 / (2 * a + 1)) ^ (2 * k + 1))
+  (log ((a + 1) / a)) :=
+ begin
+  have h₁ : |1 / (2 * a + 1)| < 1,
+  { rw [abs_of_pos, div_lt_one],
+    { linarith, },
+    { linarith, },
+    { refine div_pos one_pos (by linarith), }, },
+  { rw log_succ_div_eq_log_sub h,
+    exact has_sum_log_sub_log_of_abs_lt_1 h₁, },
```

Comment:
Don't enclose this one in braces (it was the "main goal" before the `have` was stated).

Reply:
Done